hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -208,23 +208,20 @@ public Response put(
 
       return Response.ok().status(HttpStatus.SC_OK)
           .build();
-    } catch (IOException ex) {
```

Comment:
Why do you want to change IOE to OME? Not logging for known OME makes sense, but now any IOE will get propagated directly without logging. If I catch the intention was to not log the know client issues, like permissions and stuff. But IOE can surface from any reason

Reply:
Thanks @ayushtkn for the question.

All other methods in `ObjectEndpoint` only catch `OMException` and directly propagate `IOException`.

```
$ grep -F 'catch (' hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (OMException ex) {
    } catch (UnsupportedEncodingException e) {
    } catch (IllegalArgumentException ex) {
    } catch (ParseException e) {
```

Uncaught `IOException` is handled at a different layer, and logged with stack trace:

```
s3g_1       | 2022-02-07 10:11:02,633 [qtp1498438472-16] WARN server.HttpChannel: handleException /bucket-ozone-test-2711154630/ozone-test-4349237846/putobject/key%3Dvalue/f1 java.io.IOException: TESTING
...
s3g_1       | Caused by: java.io.IOException: TESTING
s3g_1       | 	at org.apache.hadoop.ozone.s3.endpoint.ObjectEndpoint.put(ObjectEndpoint.java:200)
...
```

Reply:
Makes sense. Should be Ok then